### PR TITLE
fix(8502): Fix staging upgrade momentarily showing warning

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

- increases interval before warning the user that the upgrade might have issues
- handles case where nginx sends a 502
image
- prevents upgrade interruption when state is completing

medic/cht-core#8502

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8502-increase-interval/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8502-increase-interval/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8502-increase-interval/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by dianabarsan** [comment]
Hi @jkuester 
Would you mind having a look at this? 

Also @1yuv , I would appreciate if you could test this. You'll need to install this branch and then upgrade to another version to see it fixed, and not vice versa.

**Comment by 1yuv** [comment]
Hi @dianabarsan , I tested this with instance having fewer data and I hope it works for all scenarios. 

Can we hide this modal dialog 

image

and display information in the background like we do already ?

image

**Comment by dianabarsan** [comment]
Thanks for testing @1yuv . 

> Can we hide this modal dialog

We can. Please submit another issue for this.

File: admin/src/js/controllers/upgrade.js [hunk 1]
```diff
@@ -223,9 +223,9 @@ angular.module('controllers').controller('UpgradeCtrl',
       return $http
         .post(url, { build })
         .catch(err => {
-          // todo which status do we get with nginx???
           // exclude "50x" like statuses that come from nginx
-          if (err && (!err.status || err.status === 503 || err.status === -1) && action === 'complete') {
+          const requestFailedStatuses = [1, 502, 503];
```
Comment (jkuester) [inline admin/src/js/controllers/upgrade.js:227]: Did you mean to change `-1` to `1` here?  (I am not really sure what the case might be where the status would be either value, but seemed like the change might not have been intentional....)

**Review by jkuester — APPROVED** [review]
:+1:

File: admin/src/js/controllers/upgrade.js (hunk 1 above)
Comment (dianabarsan) [reply admin/src/js/controllers/upgrade.js:227]: oh! good spotting. It's supposed to be -1!

**Pull request merged**